pipelined/bevy_pbr2/src/render/light.rs

```diff
@@ -340,7 +533,9 @@ pub fn extract_lights(
                 shadow_normal_bias: directional_light.shadow_normal_bias
                     * directional_light_texel_size
                     * std::f32::consts::SQRT_2,
-            });
+            },
+            visible_entities.clone(),
```

Comment:
This is a potentially big list. Maybe we could just `mem::take` it? Is having last frame's visible entities around "worth" big clones every frame? Of course if we take it, we need to allocate a new list next frame, but thats still cheaper than a clone.

Reply:
Do you have an example of doing that elsewhere? Just take the inner entities vec into a local variable and assign an empty vec in its place then add a visible entities to the render world entity constructed by moving the taken entities vec into it? I’ve never used mem::take before.

Reply:
Heres an example: https://github.com/bevyengine/bevy/blob/fde5d2fe46b7a5d75660aa2cba97bbe67071592e/pipelined/bevy_render2/src/render_resource/pipeline_cache.rs#L203

`mem::take` takes the value at the given pointer, returns it, and puts the Default value in the original location.

Reply:
Nice. Done.